packages/ng/button/button.component.ts

```diff
@@ -37,13 +39,39 @@ export class ButtonComponent implements OnChanges {
 	 */
 	luButton: '' | 'outlined' | 'text' | 'text-invert' = '';
 
+	#iconComponentRef?: ElementRef<HTMLElement>;
+
+	@ContentChild(IconComponent, { read: ElementRef<HTMLElement> })
+	set iconComponentRef(ref: ElementRef<HTMLElement>) {
+		this.#iconComponentRef = ref;
+		this.updateClasses();
+	}
+
+	private get iconOnly(): boolean {
+		const listOfNode = Array.from(this.#elementRef?.nativeElement?.childNodes || []);
+		const noText = listOfNode.every((node) => node.nodeName !== '#text');
+		// ignore icon and comment
+		const noSpan =
+			listOfNode.filter((node) => {
+				return !(node.nodeName === '#comment' || (node as HTMLElement)?.tagName?.toLowerCase() === 'lu-icon');
```

Comment:
```suggestion
				return !(node.nodeName === '#comment' || node.nodeName?.toLowerCase() === 'lu-icon');
```

does this work ? According to the documentation, tagName is equal to nodeName for Element nodes. https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName

>For [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element) objects, the value of tagName is the same as the value of the [nodeName](https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeName) property the element object inherits from [Node](https://developer.mozilla.org/en-US/docs/Web/API/Node).

Reply:
(if this works we could also destruct nodeName, see previous comment)